### PR TITLE
Use Angular component syntax and hovercard updates

### PR DESCRIPTION
**Pull request by mattcollier**

### CODE REVIEW

File: session-hovercard-component.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+      onSignOut: '&brOnSignOut',
+      onRefresh: '&brOnRefresh'
+    },
+    controller: Ctrl,
+    templateUrl: requirejs.toUrl(
+      'bedrock-angular-session/session-hovercard-component.html'),
+    transclude: {
+      'action-slot': "?brSessionHovercardActionSlot",
+      'identity-slot': "?brSessionHovercardIdentitySlot",
+      'modal-slot': "?brSessionHovercardModalSlot"
```
Comment (dlongley) [inline session-hovercard-component.js]: We're not using a "slot" naming convention so, at least for the public interface, we should drop the "Slot" suffix. Also, I don't think we should have an "Identity" slot (or an "Action" slot) at all -- it's too prescriptive for how a hovercard should be used. I think it should just be `brSessionHovercardContent`. I don't understand the modal slot either ... can't whomever is providing the content just include a modal if they want? What's the extra slot for?

File: session-logout-component.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+ */
+define(['angular'], function(angular) {
+
+'use strict';
+
+function register(module) {
+  module.component('brSessionLogout', {
+    controller: Ctrl,
+    bindings: {
+      callback: '&?brLogoutCallback'
```
Comment (dlongley) [inline session-logout-component.js]: Let's make this `brOnLogout` instead, to follow our naming convention.

**Comment by dlongley** [comment]
Some comments to address/discuss.

File: session-logout-component.js (hunk 2 above)
Comment (mattcollier) [inline session-logout-component.js]: Be advised, this is a breaking change which means this would be a semver major release.

**Pull request merged**